tests/robustness/snapmeta/simple.go

```diff
@@ -28,24 +29,45 @@ func NewSimple() *Simple {
 	}
 }
 
+// IndexOperation defines several operations implemented on Index.
+type IndexOperation string
+
+const (
+	// AddToIndexOperation add value in Index.
+	AddToIndexOperation IndexOperation = "addToIndex"
+	// RemoveFromIndexOperation removes value from Index.
+	RemoveFromIndexOperation IndexOperation = "removeFromIndex"
+)
+
 // Store implements the Storer interface Store method.
-func (s *Simple) Store(key string, val []byte) error {
+func (s *Simple) Store(key string, val []byte, indexUpdates map[string]IndexOperation) error {
 	buf := make([]byte, len(val))
+
 	_ = copy(buf, val)
 
 	s.mu.Lock()
 	defer s.mu.Unlock()
 
 	s.Data[key] = buf
 
+	s.processIndexUpdates(key, indexUpdates)
+
 	return nil
 }
 
+func (s *Simple) processIndexUpdates(key string, indexUpdates map[string]IndexOperation) {
+	for indexName, indexOp := range indexUpdates {
+		switch indexOp {
+		case AddToIndexOperation:
+			s.Idx.AddToIndex(key, indexName)
+		case RemoveFromIndexOperation:
+			s.Idx.RemoveFromIndex(key, indexName)
+		}
+	}
+}
+
 // Load implements the Storer interface Load method.
 func (s *Simple) Load(key string) ([]byte, error) {
-	s.mu.Lock()
-	defer s.mu.Unlock()
-
```

Comment:
You will need these locks to protect the shared data structure from concurrent access